superbench/common/utils/gen_traffic_pattern_config.py

```diff
@@ -45,7 +79,43 @@ def __convert_config_to_host_group(config, host_list):
     return host_groups
 
 
-def gen_tarffic_pattern_host_group(host_list, pattern):
+def gen_ibstat(ansible_config, ibstat_path):    # pragma: no cover
+    """Generate the ibstat file in specified path.
+
+    Args:
+        ansible_config (DictConfig): Ansible config object.
+        ibstat_path (str): the expected path of ibstat file.
+
+    Returns:
+        ibstat_path (str): the generated path of ibstat file.
+    """
+    from superbench.runner import AnsibleClient
+    ibstat_list = []
+    stdout_regex = re.compile(r'\x1b(\[.*?[@-~]|\].*?(\x07|\x1b\\))')
+    ansible_client = AnsibleClient(ansible_config)
+    cmd = 'cat /sys/class/infiniband/*/sys_image_guid | tr -d :'
+
+    # callback function to collect and parse ibstat
+    def _ibstat_parser(artifact_dir):
+        raw_outputs = open(os.path.join(artifact_dir, 'stdout'), 'r')
```

Comment:
unify `os.path` to `Path`? and it's better to wrap in `with`, there is no `close` after `open`

Reply:
addressed